extensions/vscode/src/api/types/error.ts

```diff
@@ -1,10 +1,38 @@
 // Copyright (C) 2023 by Posit Software, PBC.
 
-export type AgentError = {
+type AgentErrorBase = {
   code: string;
   msg: string;
   operation: string;
+};
+
+export type AgentError = AgentErrorTypeUnknown | AgentErrorInvalidTOML;
+
+export type AgentErrorTypeUnknown = AgentErrorBase & {
   data: {
     [key: string]: unknown;
   };
 };
+
+// This represents the Agent Errors we haven't been able to
+// narrow down to a specific type.
+export const isAgentErrorTypeUnknown = (
+  e: AgentError,
+): e is AgentErrorTypeUnknown => {
+  return !isAgentErrorInvalidTOML(e);
+};
+
+export type AgentErrorInvalidTOML = AgentErrorBase & {
+  data: {
+    problem: string;
+    file: string;
+    line: number;
+    column: number;
+  };
+};
+
+export const isAgentErrorInvalidTOML = (
```

Comment:
With these new AgentError typing helpers we now have 3 files handling incoming agent errors:
1. `utils/errorTypes.ts` for incoming agent errors via HTTP responses (AxiosErrors)
2. `eventErrors.ts` for incoming agent errors via SSE - EventStreamMessages
3. This one

It'd be great if we can consolidate or minimize areas handling errors in different ways, I don't have a suggestion right away, but it might be worth sharing ideas to prevent confusion in the future.

Is `home.selectedConfiguration.error` an event stream error?

Reply:
No, it is present within the configuration object returned from the configuration query API. Those objects are either `Configurations` or `ConfigurationErrors`.  `ConfigurationErrors` have an `AgentError`, which is present.

The base `AgentError` object is defined within the `error.ts` file. It is the core concept. It feels like to me that we have the concept of types of `AgentErrors`, in which we have some types which include transport-specific info.

I totally agree that we need to find a good way to consolidate these. Maybe we can hop on a call real quick?

Reply:
After a quick discussion we agreed that a bit of error code consolidation would be great looking to our more complicated error handling (like this one) and trying to unify how we handle errors.

We both agreed that would be a good clean up task for a milestone in the future. After we work with our error patterns a bit more, and know which to unify on.